lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -345,8 +346,50 @@ fromMaxTxSize :: Natural -> Quantity "byte" Word16
 fromMaxTxSize =
     Quantity . fromIntegral
 
-fromPParams :: SL.PParams -> W.TxParameters
-fromPParams pp = W.TxParameters
+fromPParams :: SL.PParams -> W.ProtocolParameters
+fromPParams pp = W.ProtocolParameters
+    { decentralizationLevel =
+        decentralizationLevelFromPParams pp
+    , txParameters =
+        txParametersFromPParams pp
+    }
+
+-- | Extract the current network decentralization level from the given set of
+--   protocol parameters.
+--
+-- According to the Design Specification for Delegation and Incentives in
+-- Cardano, the decentralization parameter __/d/__ is a value in the range
+-- '[0, 1]', where:
+--
+--   * __/d/__ = '1' indicates that the network is /completely federalized/.
+--   * __/d/__ = '0' indicates that the network is /completely decentralized/.
+--
+-- However, in Cardano Wallet, we represent the decentralization level as a
+-- percentage, where:
+--
+--   * '  0 %' indicates that the network is /completely federalized/.
+--   * '100 %' indicates that the network is /completely decentralized/.
+--
+decentralizationLevelFromPParams
+    :: SL.PParams
+    -> W.DecentralizationLevel
+decentralizationLevelFromPParams pp =
+    either reportInvalidValue W.DecentralizationLevel
+        $ mkPercentage
+        $ (1 -)
+        $ SL.intervalValue d
+  where
+    d = SL._d pp
+    reportInvalidValue = error $ mconcat
```

Comment:
What about adding a `HasCallStack` constraint to this function and its callers?

Reply:
Fixed in 4bc118bf324f63f183be82c85731e4042d3c87f5.